MariaDB/server
# MDEV-30843: Add Extended descriptions to compat packages

- [x] *The Jira issue number for this PR is: MDEV-30843*


## Description
Currently Debian package: `mariadb-client-compat` and `mariadb-server-compat` are missing Extended description which leads to Lintian error:
 * extended-description-is-empty

Commit orders Description to same as rest of the Debian control file does.

Commit also fix Debian Lintian error:
 * obsolete-relation-form-in-source

## How can this PR be tested?
Build with `debian/autobake-debs.sh` and run `lintian *.deb`. There should not be:
 * E: mariadb-client-compat: extended-description-is-empty
 * E: mariadb-server-compat: extended-description-is-empty
 * E: mariadb source: obsolete-relation-form-in-source (in section for mariadb-client-compat) Conflicts mariadb-client (< 11.0.0) [debian/control:563]
 * E: mariadb source: obsolete-relation-form-in-source (in section for mariadb-client-compat) Conflicts mariadb-client-core (< 11.0.0) [debian/control:563]
 * E: mariadb source: obsolete-relation-form-in-source (in section for mariadb-client-compat) Conflicts mariadb-server (< 11.0.0) [debian/control:563]
 * E: mariadb source: obsolete-relation-form-in-source (in section for mariadb-client-compat) Conflicts mariadb-server-core (< 11.0.0) [debian/control:563]
 * E: mariadb source: obsolete-relation-form-in-source (in section for mariadb-server-compat) Conflicts mariadb-server (< 11.0.0) [debian/control:857]
 * E: mariadb source: obsolete-relation-form-in-source (in section for mariadb-server-compat) Conflicts mariadb-server-core (< 11.0.0) [debian/control:857]

## Basing the PR against the correct MariaDB version

## Backward compatibility
Should not affect make more compatible with Debian packaging rules

## PR quality check
